lib/util/index.coffee

```diff
@@ -43,45 +43,15 @@ exports.treatAsMissingDevice = (uuidOrId) ->
 		replacementErr.stack = err.stack
 		throw replacementErr
 
-safeSemver = (version) ->
-	version.replace(/(\.[0-9]+)\.rev/, '$1+rev')
-
 exports.osVersionRCompare = (versionA, versionB) ->
-	versionA = safeSemver(versionA)
-	versionB = safeSemver(versionB)
-	semverResult = semver.rcompare(versionA, versionB)
-	if semverResult != 0
-		return semverResult
-
-	revA = getRev(versionA)
-	revB = getRev(versionB)
-
-	if revA isnt revB
-		return revB - revA
-
-	devA = exports.isDevelopmentVersion(versionA)
-	devB = exports.isDevelopmentVersion(versionB)
-	if devA isnt devB
-		return devA - devB
-
-	return versionA.localeCompare(versionB)
+	return semver.rcompare(versionA, versionB)
```

Comment:
This function is now completely equivalent to `rSemver.rcompare`, so we might as well just remove it and use that directly everywhere.